src/main/java/com/schoolmanagement/schoolmanagement/controller/UserController.java

```diff
@@ -27,8 +32,12 @@ public String allAccess() {
 
     @GetMapping("/currentUser")
     @PreAuthorize("hasRole('ADMIN')")
-    public CurrentUser findByUserName() {
-        return this.securityService.currentUser();
+    public ResponseEntity<ApiResponse<CurrentUser>> findByUserName() {
+        try {
+            return ok(new ApiResponse<>(this.securityService.currentUser()));
+        } catch (final Exception e) {
+            return ok(new ApiResponse<>(new ApiError(e.getMessage())));
```

Comment:
Instead of catching generic exception and passing system generated exception message as argument to ApiError object, , it is better to handle specific exception an pass custom exception message.